datasets/customDataset.py

```diff
@@ -0,0 +1,171 @@
+import click
+import json
+import sys
+from pathlib import Path
+import os
+import numpy as np
+
+
+class CustomDataset:
+    def __init__(self):
+        self.DIR = Path(__file__).parent.resolve()
+        self.val_ratio = 0.2
+
+    def computeJsonAnswers(self, answersPath, metadataPath, outputname):
+        with open(answersPath, "r") as f:
+            answers = json.load(f)
+        with open(
+            outputname,
+            "w",
+        ) as answ:
+            json.dump(answers, answ, ensure_ascii=False, indent=3)
+
+    def computeInvertJsonAnswers(self, answersPath, metadataPath, outputname):
+        with open(answersPath, "r") as f:
+            answers = json.load(f)
+        workerAnswers = {}
+        for task in answers:
+            for worker in answers[task]:
+                workerAnswers[worker][task] = answers[task][worker]
+
+        with open(
+            outputname,
+            "w",
+        ) as answ:
+            json.dump(answers, answ, ensure_ascii=False, indent=3)
+
+    def computeRodriguezAnswers(self, answersPath, outputname):
+        crowdlabels = np.loadtxt(answersPath)
+        res_train = {task: {} for task in range(crowdlabels.shape[0])}
+        for id_, task in enumerate(crowdlabels):
+            where = np.where(task != -1)[0]
+            for worker in where:
+                res_train[id_][int(worker)] = int(task[worker])
+        with open(
+            outputname,
+            "w",
+        ) as answ:
+            json.dump(res_train, answ, ensure_ascii=False, indent=3)
+
+    def writeSymlink(self, folder, file, currentPath, position):
+        parent = file.parent.name
+        currentFile = currentPath / folder / parent / f"{file.stem}-{position}.jpg"
+        reversedParents = (currentFile.parents)[1::-1]
+
+        for subfolder in reversedParents:
+            os.makedirs(subfolder, exist_ok=True)
+        if currentFile.exists():
+            return
+        os.symlink(
+            file.absolute(),
+            currentPath / folder / parent / f"{file.stem}-{position}.jpg",
+        )
+
+    def setfolders(
+        self,
+        answers_format,
+        answers,
+        metadata,
+        label_names,
+        files_path,
+        train_path,
+        test_ground_truth,
+        test_path,
+        val_path,
+    ):
+        answersPath = Path(answers)
+
+        if answers_format == 0:
+            self.computeRodriguezAnswers(answersPath, "answers.json")
+        elif answers_format == 1:
+            if metadata == "":
+                click.echo("Please provide a valid metadata file")
+                sys.exit(1)
+            else:
+                metadataPath = Path(metadata)
+                self.computeJsonAnswers(answersPath, metadataPath, "answers.json")
+        elif answers_format == 2:
+            if metadata == "":
+                click.echo("Please provide a valid metadata file")
```

Comment:
Raise a ValueError with the message instead of click.echo followed by sys.exit(1) (better to debug in general)